keyboards/irene/keymaps/default/keymap.c

```diff
@@ -64,76 +66,106 @@ const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt)
+	if ((record->event.pressed) && (idle_mode == 1)) {
+		idle_mode = 0;
+		timer_loop = 0;
+		lcd_clrscr();
+		timer1_stop();
+	}
+
     switch (keycode) {
         case MO(_RS):
-	case TO(_QW):
-	    if (record->event.pressed) {
-		if (i == 2)
-		    i = 0;
-		else
-		    i++;
-	    }
-	    else {
-		if ((i != 2) && (i != 0))
-		    i--;
-	    }
-	    break;
-	case TG(_LW):
-	    if (record->event.pressed) {
-	    	i++;
-	    }
-	    break;
-	case KC_NO:
-	    if (record->event.pressed) {
-	    	if (lcd_off) {
-	    	    lcd_init(LCD_DISP_ON);
-		    lcd_clrscr();
-		    lcd_puts("LCD ON");
-	            lcd_off = false;
-	    	}
-	    	else {
-		    lcd_clrscr();
-		    lcd_sleep(YES);
-		    lcd_off = true;
-	    	}
-	    }
-	    return true;
-	default:
-	    return true;
+		case TO(_QW):
+			if (record->event.pressed) {
+			if (i == 2)
+				i = 0;
+			else
+				i++;
+			}
+			else {
+			if ((i != 2) && (i != 0))
+				i--;
+			}
+			break;
+		case TG(_LW):
+			if (record->event.pressed) {
+				i++;
+			}
+			break;
+		case KC_NO:
+			if (record->event.pressed) {
+				if (lcd_off) {
+					lcd_init(LCD_DISP_ON);
+				lcd_clrscr();
+				lcd_puts("LCD ON");
+					lcd_off = false;
+				}
+				else {
+				lcd_clrscr();
+				lcd_sleep(YES);
+				lcd_off = true;
+				}
+			}
+			return true;
+		case KC_F13:
+			if (record->event.pressed) {
+				if (idle_mode == 0) {
+					idle_mode = 1;
+				}
+			}
+			break;
+		default:
+			return true;
     }
     if (!lcd_off){	
+		if (idle_mode == 1) {
+			lcd_clrscr();
+			lcd_puts("Fishing...");
+			lcd_gotoxy(0,2);
+			lcd_puts_p(PSTR("     ><(((*>"));
+			lcd_gotoxy(0,4);
+			lcd_puts_p(PSTR("          <*)))><"));
+			return true;
+		}
     	switch(i){
-	    case 0:
-	    	lcd_clrscr();
-	    	lcd_puts("Layer 1");
-		lcd_gotoxy(0,2);
-		lcd_puts_p(PSTR("QWERTY"));
-		lcd_gotoxy(0,4);
-		lcd_puts_p(PSTR("(> v <)"));
-	    	break;
-	    case 1:
-	    	lcd_clrscr();
-	    	lcd_puts("Layer 2");
-		lcd_gotoxy(0,2);
-		lcd_puts_p(PSTR("Functions & Symbols"));
-		lcd_gotoxy(0,4);
-		lcd_puts_p(PSTR("(O_o)?"));
-	    	break;
-	    case 2:
-	    	lcd_clrscr();
-	    	lcd_puts("Layer 3");
-		lcd_gotoxy(0,2);
-		lcd_puts_p(PSTR("Number Pad"));
-		lcd_gotoxy(0,4);
-		lcd_puts_p(PSTR("(-_-;)"));
-	    	break;
-	    default:
-	    	break;
+			case 0:
+				lcd_clrscr();
+				lcd_puts("Layer 1");
+				lcd_gotoxy(0,2);
+				lcd_puts_p(PSTR("QWERTY"));
+				lcd_gotoxy(0,4);
+				lcd_puts_p(PSTR("(> v <)"));
+				break;
+			case 1:
+				lcd_clrscr();
+				lcd_puts("Layer 2");
+				lcd_gotoxy(0,2);
+				lcd_puts_p(PSTR("Functions & Symbols"));
+				lcd_gotoxy(0,4);
+				lcd_puts_p(PSTR("(O_o)?"));
+				break;
+			case 2:
+				lcd_clrscr();
+				lcd_puts("Layer 3");
+				lcd_gotoxy(0,2);
+				lcd_puts_p(PSTR("Number Pad"));
+				lcd_gotoxy(0,4);
+				lcd_puts_p(PSTR("(-_-;)"));
+				break;
+			default:
+				break;
         }
     }
     return true;
 }
 
+void press_random(void) {
```

Comment:
A better function name is in order. Perhaps `send_random_benign` or `send_random_benign_keystroke`.